packages/video_player/video_player_web/lib/src/video_player.dart

```diff
@@ -202,9 +203,57 @@ class VideoPlayer {
     return Duration(milliseconds: (_videoElement.currentTime * 1000).round());
   }
 
+  /// Sets options
+  Future<void> setOptions(VideoPlayerWebOptions options) async {
+    // incase called multiple times, reset options
+    _resetOptions();
+
+    if (options.controls.enabled) {
+      _videoElement.controls = true;
+      final String controlsList = options.controls.controlsList;
+      if (controlsList.isNotEmpty) {
+        _videoElement.setAttribute('controlsList', controlsList);
+      }
+
+      if (!options.controls.allowPictureInPicture) {
+        _videoElement.setAttribute('disablePictureInPicture', true);
+      }
+    }
+
+    if (!options.allowContextMenu) {
+      _onContextMenu = (html.Event event) => event.preventDefault();
+      _videoElement.addEventListener('contextmenu', _onContextMenu);
+    }
+
+    if (!options.allowRemotePlayback) {
+      _videoElement.setAttribute('disableRemotePlayback', true);
+    }
+  }
+
+  void _resetOptions() {
+    _videoElement.controls = false;
+    if (_videoElement.hasAttribute('controlsList')) {
+      _videoElement.setAttribute('controlsList', '');
```

Comment:
Instead of setting these to a weird value, I think it is more correct to `removeAttribute`. I also think you don't need to check if the attribute exists to remove it, so this method could end up much simpler, like this:

```dart
_videoElement.controls = false;
_videoElement.removeAttribute('controlsList');
_videoElement.removeAttribute('disablePictureInPicture');
_videoElement.removeAttribute('disableRemotePlayback');
if (_onContextMenu != null) {
  // Unsubscribe!
}
```

Reply:
Tests seem happy, nice simplification.